packages/api/src/brevo-config/brevo-config.resolver.ts

```diff
@@ -0,0 +1,97 @@
+import { AffectedEntity, RequiredPermission, validateNotModified } from "@comet/cms-api";
+import { EntityManager, EntityRepository, wrap } from "@mikro-orm/core";
+import { InjectRepository } from "@mikro-orm/nestjs";
+import { Type } from "@nestjs/common";
+import { Args, ID, Mutation, Query, Resolver } from "@nestjs/graphql";
+
+import { BrevoApiSenderService } from "../brevo-api/brevo-api-sender.service";
+import { BrevoApiSender } from "../brevo-api/dto/brevo-api-sender";
+import { EmailCampaignScopeInterface } from "../types";
+import { DynamicDtoValidationPipe } from "../validation/dynamic-dto-validation.pipe";
+import { BrevoConfigInput, BrevoConfigUpdateInput } from "./dto/brevo-config.input";
+import { BrevoConfigInterface } from "./entities/brevo-config-entity.factory";
+
+export function createBrevoConfigResolver({
+    Scope,
+    BrevoConfig,
+}: {
+    Scope: Type<EmailCampaignScopeInterface>;
+    BrevoConfig: Type<BrevoConfigInterface>;
+}): Type<unknown> {
+    @Resolver(() => BrevoConfig)
+    @RequiredPermission(["brevo-newsletter-config"])
+    class BrevoConfigResolver {
+        constructor(
+            private readonly entityManager: EntityManager,
+            private readonly brevoSenderApiService: BrevoApiSenderService,
+            @InjectRepository(BrevoConfig) private readonly repository: EntityRepository<BrevoConfigInterface>,
+        ) {}
+
+        private async isValidSender({ email, name }: { email: string; name: string }): Promise<boolean> {
+            const senders = await this.brevoSenderApiService.getSenders(Scope);
+
+            if (senders && senders.some((sender) => sender.email === email && sender.name === name)) {
+                return true;
+            }
+
+            return false;
+        }
+
+        @RequiredPermission(["brevo-newsletter-config"], { skipScopeCheck: true })
+        @Query(() => [BrevoApiSender], { nullable: true })
+        async senders(): Promise<Array<BrevoApiSender> | undefined> {
+            const senders = await this.brevoSenderApiService.getSenders(Scope);
+            return senders;
+        }
+
+        @Query(() => BrevoConfig, { nullable: true })
+        async brevoConfig(
+            @Args("scope", { type: () => Scope }, new DynamicDtoValidationPipe(Scope))
+            scope: typeof Scope,
+        ): Promise<BrevoConfigInterface | null> {
+            const brevoConfig = await this.repository.findOne({ scope });
+            return brevoConfig;
+        }
+
+        // TODO: add validation if the input contains a valid sender
```

Comment:
Should this be done in this MR or will there be a follow up?

Reply:
you already defined `isValidSender`

Reply:
Added here: cc8900cc784b81ede87a2899dd4ca671cf82493a

Reply:
This should also be checked on update

Reply:
Of course, sorry! Added it here: afc700525b410f25a02439378d0fb6399f565118